### ISSUE TITLE
Can't detect Dialog, Bottom Sheet, Overlay OR it also detect widgets below the Overlay

### ISSUE THREAD

**Issue by AngDrew** [fzyzcjy/flutter_convenient_test#375]
**Describe the bug**
The test can't detect  object above the screen such as Dialog, Bottom Sheet, Overlay which has their own context detached from the MaterialApp context. if it's possible to detect, the screen below the Dialog, Bottom Sheet, Overlay will still visible to the tester

**To Reproduce**
1. Make a Widget that show screen (Scaffold, with Button is fine)
2. Run the test to click the Button.
3. Show dialog or use overlay or showBottomSheet on Button pressed
4. make the test to read the Text above the Dialog / Overlay / Bottom Sheet

**Expected behavior**
the finder should not be able to see the widget below the Dialog, Bottom Sheet, Overlay

**Screenshots**
![image](https://github.com/fzyzcjy/flutter_convenient_test/assets/20359911/4aed4eb4-6e08-4c87-8687-3040e1e0c9bc)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome
 - Version 122.0.6261.112 (Official Build) (64-bit)

**Smartphone (please complete the following information):**
 - Device: Emulator
 - OS: Android API 34
 - Browser none
 - Version API Level 34

**Additional context**
is there something that i missed? in the screen shot, you can see there is too many "start date" while there is only one above my bottom sheet. there's many "start date" below the bottom sheet. is there work around for me to check if there's bottom sheet opened?

thanks!

**Comment by AngDrew** [comment]
indeed i have many animation on this app, maybe this cause some issue to the finder. anyway ill try empty screen with dialog opened using onInit() and "expand" the same error log ill try to reproduce the test first

**Comment by AngDrew** [comment]
tried to print using `find.text('Contract').allCandidates.length` and it prints 869 widgets wihle dialog are open
![image](https://github.com/fzyzcjy/flutter_convenient_test/assets/20359911/9ca2528b-81ca-4445-8c6d-4aa7e46f8943)

**Comment by AngDrew** [comment]
i was wondering, is there any way to scroll while have `showModalBottomSheet` open?

**Comment by AngDrew** [comment]
the best way to select the widget is using find.byWidgetPredicate(...) we can query any widget, and certain attribute as we wanted
but one thing i still don't understand.. how to scroll if there's multiple scrollable in the screen 🤔 how to scroll only modal bottom sheet, how to scroll from the certain position of screen (using offside or coordinate maybe?)

**Comment by AngDrew** [comment]
I found out that if we want to specifically scroll within a scrollable widget, we can use `t.tester.scrollUntilVisible(...)` and specify the scrollable parameter with a Finder.

Now, I've encountered another issue. I have a splash screen in my app that redirects users to the dashboard if they have an access token, otherwise they are redirected to the login screen. When I run the app by executing the main code (not the main test code) and receive a token, somehow the test no longer works for me. I have to uninstall the app and restart the test from scratch without a token.

Is this a bug or did I miss something? I'm relatively new to integration testing in Flutter. In the past, I've only implemented it in small projects and I'm still trying to fully understand it.

**Comment by fzyzcjy** [comment]
So if I understand correctly, your question now is about the app token thing?

I personally do something like: during test setup (e.g. setUpAll / setUp), remove that token programmatically.

**Comment by AngDrew** [comment]
turns out the token was not the issue here..

**Comment by AngDrew** [comment]
tl;dr: i cannot restart the test, it will loop infinitely and timeout

**Comment by fzyzcjy** [comment]
Hmm, firstly try to bisect which line causes this problem (e.g. add a ton of `print`s and see). Then feel free to post the details if it is nontrivial to fix!

**Comment by AngDrew** [comment]
aight, i've found the solution. it's not the library issue, but the flutter itself. 

If you would like to scroll through dialog, you should use `dragUntilVisible` instead of `scrollUntilVisible` [reference](https://stackoverflow.com/a/67990754/12716366). that said, use `scrollUntilVisible` if it's scrollable and not blocked by ignore pointer or other pointer blocking widget, and use `dragUntilVisible` if it's overlay (dialog, bottom sheet, pop up, etc.)

thanks a lot ^^

**Comment by fzyzcjy** [comment]
You are welcome and happy to see it is solved!
